service/controller/v17patch1/resource/endpoint/update.go

```diff
@@ -77,20 +77,20 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 	return updateChange, nil
 }
 
-func ipsForUpdateChange(currentIPs []string, desiredIPs []string) []string {
+func ipsForUpdateChange(currentEndpoint *Endpoint, desiredEndpoint *Endpoint) []string {
```

Comment:
That change does not improve anything. It was like that before and I changed it to what we had now. The point is to have a consistent type interface where we only deal with IPs and nothing else. The internals of this process does not need to know anything about endpoints or whatever.

Reply:
I was doing extra tests in this function before as I was trying to create the deletion behavior, but in the end, it was not necessary. I just forget to roll this back.

Reply:
Well, then please revert the change because it is not necessary and I think it is "wrong".

Reply:
done